### PR TITLE
ci: make unit tests run in parallel

### PR DESCRIPTION
**Pull request by PouriaAmini**
### Summary

This PR adds support for running unit tests in parallel for all Apple platforms.

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-SDK-Template/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  no

### CODE REVIEW

File: .github/workflows/unit-test.yml [hunk 1]
```diff
@@ -22,32 +26,44 @@ jobs:
...
-            -sdk appletvsimulator \
-            -destination 'platform=tvOS Simulator,name=Apple TV' \
-            test
-      - name: watchOS Tests
+          case "${{ matrix.platform }}" in
+            iOS)
+              xcodebuild test \
+                -scheme Amplitude-Swift-Package \
+                -sdk iphonesimulator \
+                -destination 'platform=iOS Simulator,name=iPhone 15'
```
Comment (Mercy811) [inline .github/workflows/unit-test.yml:36]: can we add `-version` for each platform based on https://github.com/amplitude/Amplitude-Swift/blob/df08b88cca1035d36b54efa01ce0d2b4ad43419d/Package.swift#L8-L13

**Review by crleona — APPROVED** [review]
Can you do a similar treatment for the release workflow? Or, even better, just make the release workflow run these jobs.

File: .github/workflows/unit-test.yml (hunk 1 above)
Comment (PouriaAmini) [reply .github/workflows/unit-test.yml]: @Mercy811 - I think the `latest` version is sufficient for most cases. We can of course add more tests to cover older OS versions later.

Comment (crleona) [reply .github/workflows/unit-test.yml]: is latest consistent with the current behavior?

Comment (PouriaAmini) [reply .github/workflows/unit-test.yml]: yes, I believe `latest` is the current behavior since no OS version is specified in the test commands.

**Pull request merged**